src/Voron/Impl/Journal/JournalReader.cs

```diff
@@ -427,18 +471,34 @@ private bool TryReadAndValidateHeader(StorageEnvironmentOptions options, out Tra
 
                 // if (txIdDiff == 1) :
                 if (current->LastPageNumber <= 0)
+                {
+                    if (CanIgnoreDataIntegrityErrorBecauseTxWasSynced(current, options))
+                        return true;
+
                     throw new InvalidDataException("Last page number after committed transaction must be greater than 0");
+                }
             }
 
             if (hashIsValid == false)
             {
+                if (CanIgnoreDataIntegrityErrorBecauseTxWasSynced(current, options))
+                    return true;
+
                 RequireHeaderUpdate = true;
                 return false;
             }
 
             return true;
         }
 
+        private bool CanIgnoreDataIntegrityErrorBecauseTxWasSynced(TransactionHeader* currentTx, StorageEnvironmentOptions options)
+        {
+            // if we have a journal which contains transactions that has been synced and this is the case for current transaction 
+            // then we can continue the recovery regardless encountered errors
+
+            return IsAlreadySyncTransaction(currentTx) && options.IgnoreDataIntegrityErrorsOfAlreadySyncedTransactions;
```

Comment:
Here, need to (carefully) check that the _next_ tx is valid (proper id and checksum)

Reply:
Why? Next transaction can also be corrupted and that's fine I think. Then we're looking for next tx valid header (where you already commented - see my answer there):

https://github.com/ravendb/ravendb/pull/9654/files/2dd60ebbde8ee0da60f0c05559e906da9bd280c6#diff-500338b5995fcc8a61f42b513e64f35eR64-R66

and if we find it and it's _still_ a transaction which was synced then we can skip multiple corrupted transactions.

Reply:
For example, we do the recovery, last sync tx is 5:

tx 1 - ok
tx 2 - corrupted
tx 3 - corrupted
tx 4 - valid
tx 5 - valid

We found tx 2 is corrupted then we start looking for next valid transaction. We found tx 4 is valid and was synced so we can safely skip both: tx 2 and tx 3

Reply:
Added TODO comment for making sure the following won't pass: https://issues.hibernatingrhinos.com/issue/RavenDB-13940#focus=streamItem-67-40594.0-0

Reply:
Okay, make sense

Reply:
Test added in https://github.com/ravendb/ravendb/pull/9654/commits/6eb47e553813fbabcb6ccaadf0cdca75d266cbac